### ISSUE TITLE
compute_loss problem

### ISSUE THREAD

**Issue by jacobrxz** [antspy/quantized_distillation#10]
Hi, the code in compute_loss function in Loss.py has code below:

loss_distilled = nn.functional.kl_div(scores, prob_teacher,
                                                  weight=self.criterion.weight,
                                                  size_average=self.criterion.size_average)

but the nn.functional.kl_div has no argument named "weight".
How it come? thanks a lot :)

**Comment by jacobrxz** [comment]
Any suggestion?

**Comment by antspy** [comment]
It is indeed weird.. I assume this used to work in pytorch 0.3, which is the version I ran.. are you using 0.3 as well?

**Comment by jacobrxz** [comment]
> It is indeed weird.. I assume this used to work in pytorch 0.3, which is the version I ran.. are you using 0.3 as well?

Yep, however, according to [PyTorch 0.3 kl_div API](https://pytorch.org/docs/0.3.0/nn.html#torch.nn.functional.kl_div), still no "weight" argument...

**Comment by antspy** [comment]
Then I must have changed this code before publishing it on GitHub.. not sure why. Feel free to delete the weight parameter, it shouldn’t be used anyhow

**Comment by jacobrxz** [comment]
> Then I must have changed this code before publishing it on GitHub.. not sure why. Feel free to delete the weight parameter, it shouldn’t be used anyhow

Thanks a lot :), that's what I did now.. but the training has two new problems:

I'm using openNMT_WMT13.py script, now for the quantized model, say 2bits quantization, the training process will be failed because the loss become very very large within 1 epoch. 4 bits encounter the same problem but within about 3 epoch. Is it right?

Another problem is the size of the model, the quantized (say 8 bits) model saved is as large as the model without quantization, also, the calculated size of the model is not that small as we expect, say the original teacher is 300M, the 8bit student model is 130M, why? I set 'quantize_first_and_last_layer':False.

Thanks you very much :)

**Comment by antspy** [comment]
Hey,

If you don’t quantize the first and last layers the model size will be bigger than 300/4 as you could expect. I’m particular, the first embedding later is usually quite big, so that should explain the difference. 

disk size is the same because we always save in full precision, and we only compute the theoretical size in low precision.

**Comment by jacobrxz** [comment]
> Hey,
> 
> If you don’t quantize the first and last layers the model size will be bigger than 300/4 as you could expect. I’m particular, the first embedding later is usually quite big, so that should explain the difference.
> 
> disk size is the same because we always save in full precision, and we only compute the theoretical size in low precision.

Got it, thank you very much! great job!
